scheduler/scheduler-api/src/test/java/org/ow2/proactive/scheduler/common/job/factories/spi/DefaultModelJobValidatorServiceProviderTest.java

```diff
@@ -0,0 +1,336 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.common.job.factories.spi;
+
+import java.io.File;
+import java.nio.charset.Charset;
+import java.util.Collections;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.ow2.proactive.scheduler.common.exception.JobCreationException;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.DefaultModelJobValidatorServiceProvider;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.BooleanParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.DateTimeParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.DoubleParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.IntegerParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.ListParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.LongParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.ModelFromURLParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.RegexpParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.URIParserValidator;
+import org.ow2.proactive.scheduler.common.job.factories.spi.model.factory.URLParserValidator;
+import org.ow2.proactive.scheduler.common.task.ScriptTask;
+import org.ow2.proactive.scheduler.common.task.Task;
+import org.ow2.proactive.scheduler.common.task.TaskVariable;
+
+
+public class DefaultModelJobValidatorServiceProviderTest {
```

Comment:
It would be much better to unit test the low-level converters and validators. Then, testing `DefaultModelJobValidatorServiceProvider` should be quite simple and limited to 2 methods based on the current implementation.